selfdrive/car/volkswagen/interface.py

```diff
@@ -22,115 +22,41 @@ def compute_gb(accel, speed):
-      ret.wheelbase = 2.98
-
-    elif candidate == CAR.GOLF_MK7:
-      # Averages of all AU Golf variants
-      ret.mass = 1397 + STD_CARGO_KG
-      ret.wheelbase = 2.62
-
-    elif candidate == CAR.JETTA_MK7:
-      # Averages of all BU Jetta variants
-      ret.mass = 1328 + STD_CARGO_KG
-      ret.wheelbase = 2.71
-
-    elif candidate == CAR.PASSAT_MK8:
-      # Averages of all 3C Passat variants
-      ret.mass = 1551 + STD_CARGO_KG
-      ret.wheelbase = 2.79
-
-    elif candidate == CAR.TIGUAN_MK2:
-      # Average of SWB and LWB variants
-      ret.mass = 1715 + STD_CARGO_KG
-      ret.wheelbase = 2.74
-
-    elif candidate == CAR.TOURAN_MK2:
-      # Average of SWB and LWB variants
-      ret.mass = 1516 + STD_CARGO_KG
-      ret.wheelbase = 2.79
-
-    elif candidate == CAR.AUDI_A3_MK3:
-      # Averages of all 8V A3 variants
-      ret.mass = 1335 + STD_CARGO_KG
-      ret.wheelbase = 2.61
-
-    elif candidate == CAR.AUDI_Q2_MK1:
-      # Averages of all GA Q2 variants
-      ret.mass = 1205 + STD_CARGO_KG
-      ret.wheelbase = 2.61
-
-    elif candidate == CAR.SEAT_ATECA_MK1:
-      # Averages of all 5F Ateca variants
-      ret.mass = 1900 + STD_CARGO_KG
-      ret.wheelbase = 2.64
-
-    elif candidate == CAR.SEAT_LEON_MK3:
-      # Averages of all 5F Leon variants
-      ret.mass = 1227 + STD_CARGO_KG
-      ret.wheelbase = 2.64
-
-    elif candidate == CAR.SKODA_KODIAQ_MK1:
-      # Averages of all 5N Kodiaq variants
-      ret.mass = 1569 + STD_CARGO_KG
-      ret.wheelbase = 2.79
-
-    elif candidate == CAR.SKODA_OCTAVIA_MK3:
-      # Averages of all 5E/NE Octavia variants
-      ret.mass = 1388 + STD_CARGO_KG
-      ret.wheelbase = 2.68
-
-    elif candidate == CAR.SKODA_SCALA_MK1:
-      # Averages of all NW Scala variants
-      ret.mass = 1192 + STD_CARGO_KG
-      ret.wheelbase = 2.65
-
-    elif candidate == CAR.SKODA_SUPERB_MK3:
-      # Averages of all 3V/NP Scala variants
-      ret.mass = 1505 + STD_CARGO_KG
-      ret.wheelbase = 2.84
+      ret.enableBsm = 0x30F in fingerprint[0]  # SWA_01
 
+    # Required per-CAR attributes
+    ret.mass = ATTRIBUTES[candidate]["mass"] + STD_CARGO_KG
+    ret.wheelbase = ATTRIBUTES[candidate]["wheelbase"]
     ret.centerToFront = ret.wheelbase * 0.45
+    # Optional per-CAR attributes, with defaults
+    ret.steerActuatorDelay = ATTRIBUTES[candidate].setdefault("steer_actuator_delay", 0.05)  # Seems good for most MQB
+    ret.steerRatio = ATTRIBUTES[candidate].setdefault("steer_ratio", 15.6)  # Updated by params learner
+    tire_stiffness_factor = ATTRIBUTES[candidate].setdefault("tire_stiffness", 1.0)  # Updated by params learner
 
-    ret.enableCamera = True
-    ret.enableBsm = 0x30F in fingerprint[0]
+    # Tuning values, currently using the same tune for all MQB
+    # If we need to tune individual models, we'll need a dict lookup by EPS parameterization, not just CAR
+    ret.steerRateCost = 1.0
+    [ ret.lateralTuning.pid.kpBP, ret.lateralTuning.pid.kiBP,
```

Comment:
revert this

Reply:
Ok, I can undo the lateralTuning.pid line rollup. Was a remnant from when I did have a tune-table but decided against.